qt/scientific_interfaces/ISISReflectometry/GUI/Preview/InstViewModel.cpp

```diff
@@ -0,0 +1,47 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "InstViewModel.h"
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+#include "MantidQtWidgets/Common/MessageHandler.h"
+#include "MantidQtWidgets/InstrumentView/InstrumentActor.h"
+#include "MantidQtWidgets/InstrumentView/RotationSurface.h"
+#include "MantidQtWidgets/InstrumentView/UnwrappedCylinder.h"
+
+#include <memory>
+
+namespace MantidQt::CustomInterfaces::ISISReflectometry {
+
+InstViewModel::InstViewModel(std::unique_ptr<MantidWidgets::IMessageHandler> messageHandler)
+    : m_messageHandler(std::move(messageHandler)) {
+  if (!m_messageHandler)
+    m_messageHandler = std::make_unique<MantidWidgets::MessageHandler>();
+}
+
+std::unique_ptr<MantidWidgets::InstrumentActor>
+InstViewModel::createInstrumentViewActor(Mantid::API::MatrixWorkspace_sptr &workspace) const {
+  bool autoscaling = true;
+  // TODO remove hard-coded scale when we add colorbar
+  auto scaleMin = 0.0;
+  auto scaleMax = 1.0;
+
+  return std::make_unique<MantidWidgets::InstrumentActor>(workspace, *m_messageHandler, autoscaling, scaleMin,
+                                                          scaleMax);
+}
+
+void InstViewModel::notifyWorkspaceUpdated(Mantid::API::MatrixWorkspace_sptr &workspace) {
+  // TODO refactor the component info stuff into the surface constructor so we don't need to get it here
```

Comment:
Is this TODO still something we're planning on doing?

Reply:
It would be good to do longer term but I think for expediency we shouldn't worry for now, so I think it best to leave the TODO in